src/cache.js

```diff
@@ -189,32 +189,35 @@ function deleteEntry(entry) {
   gcList.add(entry);
 }
 
-function invalidateEntry(entry, clearValue, deleteValue) {
+function invalidateEntry(entry, options) {
   entry.depState = 0;
   dispatchDeep(entry);
 
-  if (clearValue) {
-    entry.value = undefined;
+  if (options.clearValue) {
```

Comment:
```suggestion
  if (options.clearValue) {
    entry.value = undefined;
```

I don't know why, but you removed the line, which clears out the value from entry. With the above change, all tests pass.